### PR TITLE
feat: support BREAKING CHANGE in body

### PR DESCRIPTION
**Pull request by bcoe**
* Add support for `BREAKING CHANGE` node in `body`.
* while I was at it, did some refactoring of how we handle `BREAKING CHANGE`, adding support for `BREAKING-CHANGE` (See: [spec](https://www.conventionalcommits.org/en/v1.0.0/#specification)).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -42,49 +42,53 @@ in sync with the written specification on conventionalcommits.org.
...
-<NBSP>          ::= "U+00A0"
+<UTF8-char>       ::= "Placeholder for UTF-8 grammar"
+<UTF8-octets>     ::= <UTF8char>+
+
+<CR>              ::= "0x000D"
+<LF>              ::= "0x000A"
+<newline>         ::= [<CR>], <LF>
+<parens>          ::= "(" | ")"
+<ZWNBSP>          ::= "U+FEFF"
+<TAB>             ::= "U+0009"
```
Comment (bcoe) [inline README.md:53]: adding the `breaking-change` node forced me to indent for consistency.

File: README.md [hunk 2]
```diff
@@ -42,49 +42,53 @@ in sync with the written specification on conventionalcommits.org.
...
-<value>         ::= <text>, <continuation>+
-                 | <text>
-<continuation> ::= <newline>, <whitespace>+, <text>
+<whitespace>      ::= <ZWNBSP> | <TAB> | <VT> | <FF> | <SP> | <NBSP> | <USP>
+
+<message>         ::= <summary>, <newline>+, <body>, <newline>*, <footer>+
+                   |  <summary>, <newline>*, <footer>+
+                   |  <summary>, <newline>*
+
+/* "!" should be added to the AST as a <breaking-change> node with the value "!" */
```
Comment (bcoe) [inline README.md:67]: Added note that `!` should be added as a `breaking-change` node to the AST (_this is already the behavior_).

File: README.md [hunk 3]
```diff
@@ -42,49 +42,53 @@ in sync with the written specification on conventionalcommits.org.
...
+<summary>         ::= <type>, "(", <scope>, ")", ["!"], ":", <whitespace>*, <text>
+                   |  <type>, ["!"], ":", <whitespace>*, <text>
+<type>            ::= <any UTF8-octets except newline or parens or ":" or "!:" or whitespace>+
+<scope>           ::= <any UTF8-octets except newline or parens>+
+<text>            ::= <any UTF8-octets except newline>*
+
+
+<body>            ::= [<any body-text except pre-footer>], <newline>, <body>*
```
Comment (bcoe) [inline README.md:75]: introduced a `<body-text>` node, which is `<text>` with an optional `BREAKING CHANGE` prefix.

File: README.md [hunk 4]
```diff
@@ -42,49 +42,53 @@ in sync with the written specification on conventionalcommits.org.
...
+                   |  [<any body-text except pre-footer>]
+/* For convenience the <breaking-change>, <separator>, <whitespace>, and
```
Comment (bcoe) [inline README.md:77]: Note for implementers that we don't want to add a `<body-text>` node to the AST.

File: README.md [hunk 5]
```diff
@@ -42,49 +42,53 @@ in sync with the written specification on conventionalcommits.org.
...
+/* "!" should be added to the AST as a <breaking-change> node with the value "!" */
+<token>           ::= <breaking-change>
+                   |  <type>, "(" <scope> ")", ["!"]
+                   |  <type>, ["!"]
+<separator>       ::= ":" | " #"
+<value>           ::= <text>, <continuation>+
+                   |  <text>
+<continuation>    ::= <newline>, <whitespace>+, <text>
+
+<breaking-change> ::= "BREAKING CHANGE" | "BREAKING-CHANGE"
```
Comment (bcoe) [inline README.md:93]: The text on conventionalcommits.org indicates that we should support `BREAKING-CHANGE`.

**Review by byCedric — APPROVED** [review]

**Pull request merged**